misk-jooq/src/main/kotlin/misk/jooq/JooqModule.kt

```diff
@@ -89,7 +88,7 @@ class JooqModule(
       lateinit var clock: Clock
       override fun get(): JooqTransacter {
         return JooqTransacter(
-          dslContext = dslContext(dataSourceServiceProvider.get(), clock, datasourceConfig)
+          dslContext = lazy { dslContext(dataSourceServiceProvider.get(), clock, datasourceConfig) }
```

Comment:
core change -- just means that the `dslContext` function won't be evaluated (and thus call `dataSourceService.get()`) until first usage which I think is completely reasonable (SchemaMigrationService will likely bear the first-time-use cost rather than any action/task)